flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/FlinkFilterAggregateTransposeRule.java

```diff
@@ -0,0 +1,207 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.rules.logical;
+
+import org.apache.flink.shaded.guava33.com.google.common.collect.ImmutableList;
+
+import org.apache.calcite.plan.Contexts;
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelOptRuleOperand;
+import org.apache.calcite.plan.RelOptUtil;
+import org.apache.calcite.plan.RelRule;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.Aggregate;
+import org.apache.calcite.rel.core.Aggregate.Group;
+import org.apache.calcite.rel.core.Filter;
+import org.apache.calcite.rel.core.RelFactories;
+import org.apache.calcite.rel.rules.AggregateFilterTransposeRule;
+import org.apache.calcite.rel.rules.CoreRules;
+import org.apache.calcite.rel.rules.TransformationRule;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexUtil;
+import org.apache.calcite.tools.RelBuilder;
+import org.apache.calcite.tools.RelBuilderFactory;
+import org.apache.calcite.util.ImmutableBitSet;
+import org.immutables.value.Value;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This rule is copied from Calcite's {@link
+ * org.apache.calcite.rel.rules.FilterAggregateTransposeRule}. Modification: - Only when the filter
+ * satisfies deterministic semantics, it will be allowed to be pushed down.
+ */
+
+/**
+ * Planner rule that pushes a {@link Filter} past a {@link Aggregate}.
+ *
+ * @see AggregateFilterTransposeRule
+ * @see CoreRules#FILTER_AGGREGATE_TRANSPOSE
+ */
+@Value.Enclosing
+public class FlinkFilterAggregateTransposeRule
+        extends RelRule<FlinkFilterAggregateTransposeRule.Config> implements TransformationRule {
+    public static final RelOptRule INSTANCE = new FlinkFilterAggregateTransposeRule(Config.DEFAULT);
+
+    /** Creates a FlinkFilterAggregateTransposeRule. */
+    protected FlinkFilterAggregateTransposeRule(Config config) {
+        super(config);
+    }
+
+    @Deprecated // to be removed before 2.0
+    public FlinkFilterAggregateTransposeRule(
+            Class<? extends Filter> filterClass,
+            RelBuilderFactory relBuilderFactory,
+            Class<? extends Aggregate> aggregateClass) {
+        this(
+                Config.DEFAULT
+                        .withRelBuilderFactory(relBuilderFactory)
+                        .as(Config.class)
+                        .withOperandFor(filterClass, aggregateClass));
+    }
+
+    @Deprecated // to be removed before 2.0
```

Comment:
Can you say more about the inclusion of these deprecated methods please and the comment - which looks out of date? I know we wanted to remove methods before Flink v2 and did not do all of them as there are comments like this left in the code; but I am unsure why we would add a deprecated method.